src/Roave/SecurityAdvisories/Flag.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+declare(strict_types=1);
+
+namespace Roave\SecurityAdvisories;
+
+final class Flag
+{
+    /**
+     * within extent of same version patch flag is of the highest priority
+     * e.g. 1.1-alpha < 1.1-beta < 1.1-rc < 1.1-stable < 1.1 < 1.1-p
+     */
+    private const PRIORITY = [
+        'patch'     => 5,
+        'p'         => 5,
+        null        => 4, // special case of clean version, e.g. 1.2.3
```

Comment:
I guess it looks a bit ugly, but when I added this `null` in the map, it helped to ditch so many useless checks.
So I think it is ugly and... elegant at the same time :sweat_smile:

Reply:
`null` is not a valid array key. This is probably adding the empty string instead.

Reply:
:thinking: I'd say it actually is a valid key, at least I'm being able to get value from array by `null`   

But you got me thinking, and yes `null` is not very good key. I'm going to change it to string value with some meaning attached.   

Thanks!

Reply:
In the end empty string was the best variant. I did push update.